libs/auth-api-lib/src/lib/resources/delegation-resources.service.ts

```diff
@@ -94,4 +169,101 @@ export class DelegationResourcesService {
         children: groupChildren.get(node.name),
       }))
   }
+
+  private wrapApiScopeQuery<T extends ApiScopeQueryOptions>(
```

Comment:
Amazing query magic 🤩

Reply:
Actually I already rewrote this code and I'm surprised to see the old code here 😱 Need to review gitup! 😂

Reply:
Found it on the delegation branch. I backported it to this branch to make the PR cleaner and speed up review of the other one.

Reply:
Really nice rewrite that improves the readability a lot.